src/biogeochem/CNVegCarbonStateType.F90

```diff
@@ -2738,36 +2683,16 @@ subroutine Summary_carbonstate(this, bounds, num_allc, filter_allc, &
     ! --------------------------------------------
     ! column level summary
     ! --------------------------------------------
-
-    call p2c(bounds, num_soilc, filter_soilc, &
-         this%totvegc_patch(bounds%begp:bounds%endp), &
-         this%totvegc_col(bounds%begc:bounds%endc))
-
-    call p2c(bounds, num_soilc, filter_soilc, &
-         this%totc_patch(bounds%begp:bounds%endp), &
-         this%totc_p2c_col(bounds%begc:bounds%endc))
-
-    do fc = 1,num_allc
-       c = filter_allc(fc)
-
-       ! total ecosystem carbon, including veg but excluding cpool (TOTECOSYSC)
-       this%totecosysc_col(c) =    &
-            soilbiogeochem_cwdc_col(c)    + &
-            soilbiogeochem_totmicc_col(c) + &
-            soilbiogeochem_totlitc_col(c) + &
-            soilbiogeochem_totsomc_col(c) + &
-            this%totvegc_col(c)
-
-       ! total column carbon, including veg and cpool (TOTCOLC)
-       this%totc_col(c) =  this%totc_p2c_col(c) + &
-            soilbiogeochem_cwdc_col(c)      + &
-            soilbiogeochem_totmicc_col(c)   + &
-            soilbiogeochem_totlitc_col(c)   + &
-            soilbiogeochem_totsomc_col(c)   + &
-            soilbiogeochem_ctrunc_col(c)
-
-    end do
-
+    if(associated(this%totvegc_patch))then
```

Comment:
Its to see if the cnveg structure is allocated, since there is no patch loop involved in this part of the code. I could change it to "if (num_soilp>0) then"  which should be the same result.

Reply:
alternatively, we could just return at the top of the routine if num_soilp==0

Reply:
Ryan will change to an if on num_soilp